packages/dd-trace/src/appsec/index.js

```diff
@@ -255,15 +256,18 @@ function onResponseWriteHead ({ req, res, abortController, statusCode, responseH
   handleResults(results, req, res, rootSpan, abortController)
 }
 
+function onResponseSetHeader ({ res, abortController }) {
+  if (isBlocked(res)) {
+    abortController?.abort()
+  }
+}
```

Comment:
No, but if application calls to `setHeader` when we have already blocked the response, it could crash:
https://github.com/nodejs/node/blob/014dad5953a632f44e668f9527f546c6e1bb8b86/lib/_http_outgoing.js#L698-L699

Reply:
oh wow i didn't know